### ISSUE TITLE
cryptokeys missing when DNSSEC-enabled zone is deleted/recreated quickly

### ISSUE THREAD

**Issue by peterthomassen** [PowerDNS/pdns#7407]
- Program: Authoritative
 - Issue type: Bug report

### Short description
All this is using the API.

- When deleting a DNSSEC-enabled zone and recreating it again within 30 seconds, cryptokeys will be missing (both via `GET` to the `cryptokeys` endpoint, and also in the `cryptokeys` table of the database).
- Trying again (by deleting and recreating once more) reliably produces a zone that has keys.
- Trying even more often leads to an alternating pattern.
- Whenever there were more than 30 seconds between the attempts, keys are created reliably.

### Environment
 - Operating system: Debian stretch
 - Software version: 4.1.5-1pdns.stretch
 - Software source: PowerDNS repository

### Steps to reproduce
1. Create a shell script with the following content (adjust hostname if it is not `pdns`):
    ```
    curl -s -X POST http://pdns:8081/api/v1/servers/localhost/zones -H "Content-Type: application/json" -H "X-API-Key: $TOKEN" -d '{"name": "test.de.", "kind": "MASTER", "dnssec": true, "nsec3param": "1 0 127 abd9404aa407e63d", "nameservers": ["ns1.test.com.", "ns2.test.com."]}' | wc
    curl -s -X GET http://pdns:8081/api/v1/servers/localhost/zones/test.de./cryptokeys -H "Content-Type: application/json" -H "X-API-Key: $TOKEN" | wc
    curl -s -X DELETE http://pdns:8081/api/v1/servers/localhost/zones/test.de. -H "Content-Type: application/json" -H "X-API-Key: $TOKEN" | wc
    ```

2. `export TOKEN=...`, then run the script with various delays
3. Observe how the second line varies (it has the response length in the last column)

Remove `| wc` to see the full (confusing) output.

### Expected behaviour
Keys should always be present, i.e. second line should never have last column equal to 2.

### Actual behaviour
Last column of second line often is 2, i.e. the `cryptokeys` endpoint returned `[]` (keys missing).

**Comment by Habbie** [comment]
```
# dnssec-key-cache-ttl	Seconds to cache DNSSEC keys from the database
# dnssec-key-cache-ttl=30
```

this default (that you did not override) looks like a smoking gun

**Comment by peterthomassen** [comment]
Setting `dnssec-key-cache-ttl=0` indeed is a workaround for this problem. Thanks for the very very quick response!

Obviously, that's not a real fix (otherwise the configuration option would be pointless). We'll go for the workaround for now; the preferred solution would be to purge keys from the cache at zone deletion time.

**Comment by Habbie** [comment]
Mitigation in #7408.

**Comment by Habbie** [comment]
Fixed in #7529
